src/python/pants/base/build_environment.py

```diff
@@ -34,7 +34,9 @@ def get_buildroot() -> str:
 
 def get_pants_cachedir() -> str:
```

Comment:
If this is only being used in BinaryUtil now, it would be good to move there. Ditto on `pants_configdir`. It reduces the risk of people accidentally using this.

Or, maybe even better, keep these, but have them return `Native().default_cache_path()`. Move this non-Rust version into `BinaryUtil`.

Reply:
It goes much deeper than that, unfortunately... follow the references from the method I linked.

Reply:
But maybe I don't understand what you're suggesting.

Reply:
My suggestion is to stop implementing any custom logic in `base/build_environment.py`. Solely have this function return `Native().default_cache_path()`. Move this implementation to `binary_util.py`.

But, actually, don't spend time on this. It's not that big of a deal because you have the new unit tests. It's not worth spending more time on this, imo.

Reply:
> Move this implementation to binary_util.py.

Yea, unfortunately that won't work: see my expanded response above.